gui.py

```diff
@@ -0,0 +1,189 @@
+                text = state.enemy_tanks[cell]
+            elif cell in map_.base:
+                color = ucf.BASE_COLOR
+            elif cell in map_.spawn_points:
+                color = ucf.SPAWN_COLOR
+            hex_ = Hex(self.hex_outer_radius, color, text)
+            hex_.setParent(self)
+            hex_.setObjectName(f"{cell}")
+            shift_x, shift_y = self.hex_to_pixel(cell)
+            hex_.move(self.get_mid_map()[0] + shift_x, self.get_mid_map()[1] + shift_y)
+            hex_.show()
+
+    def show_message(self, text: str) -> None:
+        """
+        Create pop-up message window
+        :param text: text to display
+        :return: None
+        """
+        message = QtWidgets.QMessageBox(text=text, parent=self)
+        message.show()
+
+    def closeEvent(self, event: QtGui.QCloseEvent) -> None:
+        """
+        Close event handler
+        :param event: system generated event
+        :return: None
+        """
+        self.presenter_thread.exit()
+
+
+class Hex(QtWidgets.QWidget):
+    """
+    Hexagonal Widget class used to display map cells in main window
+    """
+    def __init__(self, hex_outer_radius, color, text="", parent=None):
+        super().__init__(parent)
+        self.color = color
+        self.text = text
+        self.pen = QtGui.QPen(QtGui.QColor(*ucf.HEX_BORDER_COLOR))
+        self.pen.setWidth(ucf.HEX_BORDER_WEIGHT)
+        self.brush = QtGui.QBrush(QtGui.QColor(*self.color))
+        self.polygon = self.create_hex(hex_outer_radius)
+
+    @lru_cache(1)
+    def create_hex(self, radius: float) -> QtGui.QPolygonF:
+        """
+        Creates PySide QPolygonF obj of hexagon inscribed in a circle of given radius
+        :param radius: circumscribed circle radius
+        :return: PySide QPolygonF obj
+        """
+        polygon = QtGui.QPolygonF()
+        for point in self.get_hex_points(radius):
+            polygon.append(QtCore.QPointF(*point))
+        return polygon
+
+    @staticmethod
+    def get_hex_points(radius: float) -> Generator:
+        """
+        Creates generator of vertex coordinates in pixels relative
+        to center of hexagon inscribed in a circle of given radius
+        :param radius: circumscribed circle radius
+        :return: Generator
+        """
+        width = radius * 2
+        height = math.sqrt(3) * radius
+        return ((width / 2 + radius * math.cos(math.radians(60 * i)),
+                 height / 2 + radius * math.sin(math.radians(60 * i))) for i in range(6))
+
+    def paintEvent(self, event: QtGui.QPaintEvent) -> None:
+        """
+        Paint event handler
+        :param event: system generated event
+        :return: None
+        """
+        painter = QtGui.QPainter(self)
+        painter.setPen(self.pen)
+        painter.setBrush(self.brush)
+        painter.drawPolygon(self.polygon)
+        painter.drawText(12, 12, f"{self.text}")
+        painter.end()
+
+
+if __name__ == "__main__":
```

Comment:
If this is an execution entry point than it is better to move it to a separate module with corresponding name.